### DISCUSSION TITLE
Problem while calculating protein-protein electrostatic energy

### DISCUSSION THREAD

**Discussion by srdjan125** [Electrostatics/apbs#146] [Q&A] [answered]
Hello,

I want to calculate electrostatic energy between two proteins while their distance is varied, ie initial center-to-center distance when two proteins are in close contact is around 3nm, and final distance when two proteins are far away from each other is 30nm. 
 I want to calculate their electrostatic interaction energy as a function of their distance (Coloumb energy substracted by solvation energy due to ionic solvent screening).

  This this, I have tried binding energy calculations like in this tutorial: https://www.ics.uci.edu/~dock/manuals/apbs/html/tutorial/x307.html where I calculate energy of A, energy of B, energy of AB complex and then Interaction_energy=AB-A-B.

  I have used the same parameters like in the above link except the dime, cglen and fglen which I have determined by pysize.py python program.
   For the case when proteins are far away the parameters are:
    mg-auto
    dime 129 129 801
    cglen 62.353 70.198 647.884
    fglen 56.678 61.293 401.108

For the case when proteins are in closest contact parameters given by pysize.py are:
   mg-auto
    dime 97 129 129
    cglen 62.353 70.195 139.584
    fglen 56.678 61.291 102.108

I have understood that I should use the same grid parameters for all calculations, so I have used the first parameters for both calculations.

   However, I got some illogical result: electrostatic energy between A and B is higher when they are 30nm away then when they are 3nm away, which is counterintuitive: there should be higher repulsion energy between two proteins in close contact then in case when they are far away.

  I have tried to do the same calculations after putting salt concentration is 0, and in that case I got "correct" results: there is high electrostatic repulsive energy in close contact and low one at 30nm distance.
  From this I understand that there might be issue from electrostatic energy of salts around proteins, so this term does not cancel out. I also think it might be the problem of centering fine grid: in one case fine grid is centered on short distance AB complex, while in another case in the middle of two proteins (at 15nm distance between them), but if I place fine grid on protein A, then protein B will be 30nm away and not part of fine grid.

I can get "correct" ratio of energies when I take these parameters:
    mg-auto
    dime 129 129 801
    cglen 129 129 801
    fglen 56.678 61.293 401.108
however this size of cglen is not recommended by APBS tutorial or pysize.py program. I am new to APBS, but in case of MD it would make sense to take larger box size to cancel out local solvent effects. 

   I would appreciate if someone could help me with this issue or give some advice what might be the problem.
   Should I do calculations for full thermodynamics cycle? 
   Or instead of using lpbe I should use lrpbe to get solvation energies directly and calculate Coloumb energies separately?

**Comment by sobolevnrm** [comment]
Hello --

Thank you for the detailed description of your question.

> I have used the same parameters like in the above link except the dime, cglen and fglen which I have determined by pysize.py python program.
For the case when proteins are far away the parameters are:
mg-auto
dime 129 129 801
cglen 62.353 70.198 647.884
fglen 56.678 61.293 401.108
> 
> For the case when proteins are in closest contact parameters given by pysize.py are:
mg-auto
dime 97 129 129
cglen 62.353 70.195 139.584
fglen 56.678 61.291 102.108
> 
> I have understood that I should use the same grid parameters for all calculations, so I have used the first parameters for both calculations.

Using the first set of parameters is correct.  However, as you note below, you also need to fix the grid center with the `cgcent` and `fgcent` parameters.  I would recommend fixing these to specific coordinates rather than positions relative to molecules.  For the first case, how far are the boundaries of the mesh from the surface of the molecules?

> I have tried to do the same calculations after putting salt concentration is 0, and in that case I got "correct" results: there is high electrostatic repulsive energy in close contact and low one at 30nm distance.
> From this I understand that there might be issue from electrostatic energy of salts around proteins, so this term does not cancel out.

Yes, that seems likely.  Can you say more about the salt (concentration, valence) you're using as well as the charges on the protein?

> I can get "correct" ratio of energies when I take these parameters:
> mg-auto
> dime 129 129 801
> cglen 129 129 801
> fglen 56.678 61.293 401.108
> however this size of cglen is not recommended by APBS tutorial or pysize.py program. I am new to APBS, but in case of MD it would make sense to take larger box size to cancel out local solvent effects.

The same boundary artifacts can occur in PBE calculations as well.  What are you using for `bcfl`?

> Should I do calculations for full thermodynamics cycle?
> Or instead of using lpbe I should use lrpbe to get solvation energies directly and calculate Coloumb energies separately?

The `lrpbe` functionality in APBS is not very reliable; I would recommend trying the thermodynamic cycle with the grid settings you describe above.

Thank you

**Comment by srdjan125** [comment]
Hello,

Thank you for the system. At the moment I am testing calculations at 0.05M salt concentrations, but eventually I should run calculations at different ionic strengths (up to 1M or even higher).
   I am also testing in lysozyme (small and simple protein), at current pH it has charge +9 but I am planning to perform calculations for other proteins/pH too.
    This protein has diameter around 3.4nm , and fine grid length in z dimension is 647A (64.7 nm) which means if protein COM distance is 30nm, their edge is around 33.4nm which seems sufficient to fit into fine grid box.

  Eventually I have changed boundary conditions from sdh to mdh, and with mdh (Multiple Debye Huckel conditions) I have got more logical and satisfying results.

  Witth mdh and cg/fg grids recommended by psize.py I have got the following results:   E(complex_close)=22KJ/mol    E(complex_30nm)=0.00004KJ/mol.  By E(complex) I mean final energy (AB-A-B ).
   Before with sdh conditions and the same grid parameters I got: E(complex_close)=26KJ/mol    E(complex_30nm)=46KJ/mol.

 Before when I have used sdh and I have increased the size of coarse grid lengths (but keeping fine grid lengths recommended by psize.py) I have got the following results:  E(complex_close)=26KJ/mol    E(complex_30nm)=3KJ/mol.

  From the point of simple Couloumb or one sphere Debye Huckel theory I would rather trust value 0.00004KJ/mol than 3KJ/mol when proteins are 30nm away. (although I understand that PB includes hydration effects too).

    So I suppose that problem was in setting correct boundary conditions, although I still think that there might be still be some artifacts to cancel out the salt energies, due to the fact that proteins are pulled away and the way of setting the fine grid.

  May I ask what do I gain if I set fine grid on coordinate rather then on complex. If I set fine grid on complex (two proteins 30 nm away) the APBS will make both proeins (and solvent between them) as center of grid.
  If I want to set grid center manually, I would need to find some coordinate between two proteins (roughly 15nm away from both of them), so I will get the same.

   May I also ask another 2 short questions rather than opening new Discussion board.

a) In all tutorials I see that that cation and anion radii are set to 2A. Why I cannot set experimental radii of cations? (for example for NaCl would be something like 1.9A and 2.5A or whatever the correct values are)

b) How can I convert KJ/mol units? For example, if I want to compare calculations with some simple theory (lets say Debye Huckel) in that case I would get that energy between 2 molecules is J.
  In case of KJ/mol I need to convert KJ to 1000J and mol to Na number of proteins. But is it 1 mol of proteins or 1 mol of protein pairs?  This changes conversion by factor of 2 (whether is KJ per mole of single proteins , or KJ per mole of protein pairs - or per mole of "simulation box")


   Greetings.



________________________________

**Comment by sobolevnrm** [comment]
>  May I ask what do I gain if I set fine grid on coordinate rather then on complex. If I set fine grid on complex (two proteins 30 nm away) the APBS will make both proeins (and solvent between them) as center of grid.
  If I want to set grid center manually, I would need to find some coordinate between two proteins (roughly 15nm away from both of them), so I will get the same.
  
The grid centering based on structures is primarily a feature intended for convenience so users don't have to calculate the coordinates for the center themselves.

**Comment by sobolevnrm** [comment, marked as answer]
> May I also ask another 2 short questions rather than opening new Discussion board.

FWIW, a new discussion thread is probably easier when covering multiple topics.

**Comment by sobolevnrm** [comment]
> a) In all tutorials I see that that cation and anion radii are set to 2A. Why I cannot set experimental radii of cations? (for example for NaCl would be something like 1.9A and 2.5A or whatever the correct values are)

You can set experimental radii; however, APBS will only use the largest ion radius you assign.

**Comment by sobolevnrm** [comment]
> b) How can I convert KJ/mol units? For example, if I want to compare calculations with some simple theory (lets say Debye Huckel) in that case I would get that energy between 2 molecules is J.  In case of KJ/mol I need to convert KJ to 1000J and mol to Na number of proteins. But is it 1 mol of proteins or 1 mol of protein pairs?  This changes conversion by factor of 2 (whether is KJ per mole of single proteins , or KJ per mole of protein pairs - or per mole of "simulation box")

You will need a factor of 1000 to convert between kJ and J.  You should not need a factor of 2 to account for the number of molecules.
